app/internal/import_file.py

```diff
@@ -0,0 +1,178 @@
+                          max_event_start_date: int
+                          = MAX_EVENTS_START_DATE) -> bool:
+    """
+    checking after importing that there is no larger quantity of events
+    with the same date according to the rule we have set.
+    """
+    same_date_counter = 1
+    date_n_count = {}
+    for event in calendar1:
+        if event["Date"] in date_n_count:
+            date_n_count[event["Date"]] += 1
+            if date_n_count[event["Date"]] > same_date_counter:
+                same_date_counter = date_n_count[event["Date"]]
+        else:
+            date_n_count[event["Date"]] = 1
+    return same_date_counter <= max_event_start_date
+
+
+def import_txt_file(txt_file: str) -> List[Dict[str, Union[str, Any]]]:
+    calendar_content = []
+    with open(txt_file, "r") as text:
+        events = text.readlines()
+    for event in events:
+        if not check_validity_of_txt(event):
+            return list()
+        head, content, event_date = event.split(", ")
+        if not check_date_in_range(event_date.replace("\n", "")):
+            return list()
+        event_date = datetime.datetime.strptime(event_date.replace("\n", ""),
+                                                "%m-%d-%Y")
+        calendar_content.append({"Head": head,
+                                 "Content": content,
+                                 "Date": event_date})
+    return calendar_content
+
+
+def import_ics_file(ics_file: str) -> List[Dict[str, Union[str, Any]]]:
+    calendar_content = []
+    with open(ics_file, "r") as ics:
+        try:
+            calendar_read = Calendar.from_ical(ics.read())
+        except (IndexError, ValueError):
+            return list()
+        for component in calendar_read.walk():
+            if component.name == "VEVENT":
+                if str(component.get('summary')) is None or \
+                   component.get('dtstart') is None or \
+                   not check_date_in_range(component.get('dtstart').dt):
+                    return list()
+                else:
+                    calendar_content.append({
+                        "Head": str(component.get('summary')),
+                        "Content": str(component.get('description')),
+                        "Date": component.get('dtstart').dt
+                        .replace(tzinfo=None)
+                    })
+    return calendar_content
+
+
+def move_events_to_db(events: List[Dict[str, Union[str, Any]]],
+                      user_id: int,
+                      session: SessionLocal = SessionLocal()) -> None:
+    """insert the events into Event table"""
+    for event in events:
+        event = Event(
+            title=event["Head"],
+            content=event["Content"],
+            date=event["Date"],
+            owner_id=user_id
+        )
+        session.add(event)
+    session.commit()
+    session.close()
+
+
+def user_click_import(file: str, user_id: int,
```

Comment:
I might have missed it, but I couldn't find where this function is called other than in the tests. If this happens from a post requests, this should move to /routers, possibly /routers/event. Maybe also change the name to import_event().